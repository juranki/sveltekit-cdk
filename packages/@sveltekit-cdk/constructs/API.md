# API Reference <a name="API Reference" id="api-reference"></a>

## Constructs <a name="Constructs" id="Constructs"></a>

### SvelteDistribution <a name="SvelteDistribution" id="@sveltekit-cdk/constructs.SvelteDistribution"></a>

#### Initializers <a name="Initializers" id="@sveltekit-cdk/constructs.SvelteDistribution.Initializer"></a>

```typescript
import { SvelteDistribution } from '@sveltekit-cdk/constructs'

new SvelteDistribution(scope: Construct, id: string, props: SvelteDistributionProps)
```

| **Name** | **Type** | **Description** |
| --- | --- | --- |
| <code><a href="#@sveltekit-cdk/constructs.SvelteDistribution.Initializer.parameter.scope">scope</a></code> | <code>constructs.Construct</code> | *No description.* |
| <code><a href="#@sveltekit-cdk/constructs.SvelteDistribution.Initializer.parameter.id">id</a></code> | <code>string</code> | *No description.* |
| <code><a href="#@sveltekit-cdk/constructs.SvelteDistribution.Initializer.parameter.props">props</a></code> | <code><a href="#@sveltekit-cdk/constructs.SvelteDistributionProps">SvelteDistributionProps</a></code> | *No description.* |

---

##### `scope`<sup>Required</sup> <a name="scope" id="@sveltekit-cdk/constructs.SvelteDistribution.Initializer.parameter.scope"></a>

- *Type:* constructs.Construct

---

##### `id`<sup>Required</sup> <a name="id" id="@sveltekit-cdk/constructs.SvelteDistribution.Initializer.parameter.id"></a>

- *Type:* string

---

##### `props`<sup>Required</sup> <a name="props" id="@sveltekit-cdk/constructs.SvelteDistribution.Initializer.parameter.props"></a>

- *Type:* <a href="#@sveltekit-cdk/constructs.SvelteDistributionProps">SvelteDistributionProps</a>

---

#### Methods <a name="Methods" id="Methods"></a>

| **Name** | **Description** |
| --- | --- |
| <code><a href="#@sveltekit-cdk/constructs.SvelteDistribution.toString">toString</a></code> | Returns a string representation of this construct. |

---

##### `toString` <a name="toString" id="@sveltekit-cdk/constructs.SvelteDistribution.toString"></a>

```typescript
public toString(): string
```

Returns a string representation of this construct.

#### Static Functions <a name="Static Functions" id="Static Functions"></a>

| **Name** | **Description** |
| --- | --- |
| <code><a href="#@sveltekit-cdk/constructs.SvelteDistribution.isConstruct">isConstruct</a></code> | Checks if `x` is a construct. |

---

##### ~~`isConstruct`~~ <a name="isConstruct" id="@sveltekit-cdk/constructs.SvelteDistribution.isConstruct"></a>

```typescript
import { SvelteDistribution } from '@sveltekit-cdk/constructs'

SvelteDistribution.isConstruct(x: any)
```

Checks if `x` is a construct.

###### `x`<sup>Required</sup> <a name="x" id="@sveltekit-cdk/constructs.SvelteDistribution.isConstruct.parameter.x"></a>

- *Type:* any

Any object.

---

#### Properties <a name="Properties" id="Properties"></a>

| **Name** | **Type** | **Description** |
| --- | --- | --- |
| <code><a href="#@sveltekit-cdk/constructs.SvelteDistribution.property.node">node</a></code> | <code>constructs.Node</code> | The tree node. |
| <code><a href="#@sveltekit-cdk/constructs.SvelteDistribution.property.bucket">bucket</a></code> | <code>aws-cdk-lib.aws_s3.Bucket</code> | *No description.* |
| <code><a href="#@sveltekit-cdk/constructs.SvelteDistribution.property.distribution">distribution</a></code> | <code>aws-cdk-lib.aws_cloudfront.Distribution</code> | *No description.* |
| <code><a href="#@sveltekit-cdk/constructs.SvelteDistribution.property.function">function</a></code> | <code>aws-cdk-lib.aws_cloudfront.experimental.EdgeFunction</code> | *No description.* |

---

##### `node`<sup>Required</sup> <a name="node" id="@sveltekit-cdk/constructs.SvelteDistribution.property.node"></a>

```typescript
public readonly node: Node;
```

- *Type:* constructs.Node

The tree node.

---

##### `bucket`<sup>Required</sup> <a name="bucket" id="@sveltekit-cdk/constructs.SvelteDistribution.property.bucket"></a>

```typescript
public readonly bucket: Bucket;
```

- *Type:* aws-cdk-lib.aws_s3.Bucket

---

##### `distribution`<sup>Required</sup> <a name="distribution" id="@sveltekit-cdk/constructs.SvelteDistribution.property.distribution"></a>

```typescript
public readonly distribution: Distribution;
```

- *Type:* aws-cdk-lib.aws_cloudfront.Distribution

---

##### `function`<sup>Optional</sup> <a name="function" id="@sveltekit-cdk/constructs.SvelteDistribution.property.function"></a>

```typescript
public readonly function: EdgeFunction;
```

- *Type:* aws-cdk-lib.aws_cloudfront.experimental.EdgeFunction

---


## Structs <a name="Structs" id="Structs"></a>

### RendererProps <a name="RendererProps" id="@sveltekit-cdk/constructs.RendererProps"></a>

#### Initializer <a name="Initializer" id="@sveltekit-cdk/constructs.RendererProps.Initializer"></a>

```typescript
import { RendererProps } from '@sveltekit-cdk/constructs'

const rendererProps: RendererProps = { ... }
```

#### Properties <a name="Properties" id="Properties"></a>

| **Name** | **Type** | **Description** |
| --- | --- | --- |
| <code><a href="#@sveltekit-cdk/constructs.RendererProps.property.artifactPath">artifactPath</a></code> | <code>string</code> | Location of sveltekit artifacts. |
| <code><a href="#@sveltekit-cdk/constructs.RendererProps.property.environment">environment</a></code> | <code>{[ key: string ]: string}</code> | Environment variables for the backend implementation. |
| <code><a href="#@sveltekit-cdk/constructs.RendererProps.property.logLevel">logLevel</a></code> | <code>string</code> | Logging verbosity (default: INFO). |

---

##### `artifactPath`<sup>Optional</sup> <a name="artifactPath" id="@sveltekit-cdk/constructs.RendererProps.property.artifactPath"></a>

```typescript
public readonly artifactPath: string;
```

- *Type:* string
- *Default:* 'sveltekit'

Location of sveltekit artifacts.

---

##### `environment`<sup>Optional</sup> <a name="environment" id="@sveltekit-cdk/constructs.RendererProps.property.environment"></a>

```typescript
public readonly environment: {[ key: string ]: string};
```

- *Type:* {[ key: string ]: string}

Environment variables for the backend implementation.

---

##### `logLevel`<sup>Optional</sup> <a name="logLevel" id="@sveltekit-cdk/constructs.RendererProps.property.logLevel"></a>

```typescript
public readonly logLevel: string;
```

- *Type:* string

Logging verbosity (default: INFO).

---

### SvelteDistributionProps <a name="SvelteDistributionProps" id="@sveltekit-cdk/constructs.SvelteDistributionProps"></a>

#### Initializer <a name="Initializer" id="@sveltekit-cdk/constructs.SvelteDistributionProps.Initializer"></a>

```typescript
import { SvelteDistributionProps } from '@sveltekit-cdk/constructs'

const svelteDistributionProps: SvelteDistributionProps = { ... }
```

#### Properties <a name="Properties" id="Properties"></a>

| **Name** | **Type** | **Description** |
| --- | --- | --- |
| <code><a href="#@sveltekit-cdk/constructs.SvelteDistributionProps.property.artifactPath">artifactPath</a></code> | <code>string</code> | Location of sveltekit artifacts. |
| <code><a href="#@sveltekit-cdk/constructs.SvelteDistributionProps.property.bucketProps">bucketProps</a></code> | <code>aws-cdk-lib.aws_s3.BucketProps</code> | Bucket props for the svelteStaticBucket s3 bucket. |
| <code><a href="#@sveltekit-cdk/constructs.SvelteDistributionProps.property.cachePolicy">cachePolicy</a></code> | <code>aws-cdk-lib.aws_cloudfront.ICachePolicy</code> | Cache policy determies caching for dynamic content. |
| <code><a href="#@sveltekit-cdk/constructs.SvelteDistributionProps.property.certificateArn">certificateArn</a></code> | <code>string</code> | Certificate to use with the CloudFront Distribution. |
| <code><a href="#@sveltekit-cdk/constructs.SvelteDistributionProps.property.domainNames">domainNames</a></code> | <code>string[]</code> | Domain names to associate with the CloudFront Distribution. |
| <code><a href="#@sveltekit-cdk/constructs.SvelteDistributionProps.property.originRequestPolicy">originRequestPolicy</a></code> | <code>aws-cdk-lib.aws_cloudfront.IOriginRequestPolicy</code> | Origin request policy determines which parts of requests CloudFront passes to your backend. |
| <code><a href="#@sveltekit-cdk/constructs.SvelteDistributionProps.property.priceClass">priceClass</a></code> | <code>aws-cdk-lib.aws_cloudfront.PriceClass</code> | PriceClass. |
| <code><a href="#@sveltekit-cdk/constructs.SvelteDistributionProps.property.rendererProps">rendererProps</a></code> | <code><a href="#@sveltekit-cdk/constructs.RendererProps">RendererProps</a></code> | Props for Lambda@Edge renderer. |

---

##### `artifactPath`<sup>Optional</sup> <a name="artifactPath" id="@sveltekit-cdk/constructs.SvelteDistributionProps.property.artifactPath"></a>

```typescript
public readonly artifactPath: string;
```

- *Type:* string
- *Default:* 'sveltekit'

Location of sveltekit artifacts.

---

##### `bucketProps`<sup>Optional</sup> <a name="bucketProps" id="@sveltekit-cdk/constructs.SvelteDistributionProps.property.bucketProps"></a>

```typescript
public readonly bucketProps: BucketProps;
```

- *Type:* aws-cdk-lib.aws_s3.BucketProps

Bucket props for the svelteStaticBucket s3 bucket.

> [https://docs.aws.amazon.com/cdk/api/v1/docs/](https://docs.aws.amazon.com/cdk/api/v1/docs/)

---

##### `cachePolicy`<sup>Optional</sup> <a name="cachePolicy" id="@sveltekit-cdk/constructs.SvelteDistributionProps.property.cachePolicy"></a>

```typescript
public readonly cachePolicy: ICachePolicy;
```

- *Type:* aws-cdk-lib.aws_cloudfront.ICachePolicy
- *Default:* CACHING_DISABLED

Cache policy determies caching for dynamic content.

Note: static content is cached using default setting (CACHING_OPTIMIZED).

> [https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/controlling-origin-requests.html](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/controlling-origin-requests.html)

---

##### `certificateArn`<sup>Optional</sup> <a name="certificateArn" id="@sveltekit-cdk/constructs.SvelteDistributionProps.property.certificateArn"></a>

```typescript
public readonly certificateArn: string;
```

- *Type:* string
- *Default:* undefined

Certificate to use with the CloudFront Distribution.

---

##### `domainNames`<sup>Optional</sup> <a name="domainNames" id="@sveltekit-cdk/constructs.SvelteDistributionProps.property.domainNames"></a>

```typescript
public readonly domainNames: string[];
```

- *Type:* string[]
- *Default:* undefined

Domain names to associate with the CloudFront Distribution.

---

##### `originRequestPolicy`<sup>Optional</sup> <a name="originRequestPolicy" id="@sveltekit-cdk/constructs.SvelteDistributionProps.property.originRequestPolicy"></a>

```typescript
public readonly originRequestPolicy: IOriginRequestPolicy;
```

- *Type:* aws-cdk-lib.aws_cloudfront.IOriginRequestPolicy
- *Default:* AllViewer managed policy

Origin request policy determines which parts of requests CloudFront passes to your backend.

> [https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/controlling-origin-requests.html](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/controlling-origin-requests.html)

---

##### `priceClass`<sup>Optional</sup> <a name="priceClass" id="@sveltekit-cdk/constructs.SvelteDistributionProps.property.priceClass"></a>

```typescript
public readonly priceClass: PriceClass;
```

- *Type:* aws-cdk-lib.aws_cloudfront.PriceClass
- *Default:* PriceClass.PRICE_CLASS_100

PriceClass.

> [https://docs.aws.amazon.com/cdk/api/latest/typescript/api/aws-cloudfront/priceclass.html#aws_cloudfront_PriceClass](https://docs.aws.amazon.com/cdk/api/latest/typescript/api/aws-cloudfront/priceclass.html#aws_cloudfront_PriceClass)

---

##### `rendererProps`<sup>Optional</sup> <a name="rendererProps" id="@sveltekit-cdk/constructs.SvelteDistributionProps.property.rendererProps"></a>

```typescript
public readonly rendererProps: RendererProps;
```

- *Type:* <a href="#@sveltekit-cdk/constructs.RendererProps">RendererProps</a>

Props for Lambda@Edge renderer.

---



