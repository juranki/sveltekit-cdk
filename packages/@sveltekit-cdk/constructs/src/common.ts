export const DEFAULT_ARTIFACT_PATH = 'sveltekit';

export interface RendererProps {
  /**
   * Location of sveltekit artifacts
   *
   * @default 'sveltekit'
   */
  readonly artifactPath?: string;
  /**
   * Environment variables for the backend implementation
   */
  readonly environment?: {
    [key: string]: string;
  };
  /**
   * Logging verbosity (default: INFO)
   */
  readonly logLevel?: 'ERROR' | 'WARN' | 'INFO' | 'DEBUG';
}

export type StaticRoutes = Record<string, 'prerendered' | 'static'>
