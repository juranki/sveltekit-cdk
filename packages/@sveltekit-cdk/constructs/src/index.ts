export { RendererProps } from './common';
export { SvelteDistribution, SvelteDistributionProps } from './distribution';
