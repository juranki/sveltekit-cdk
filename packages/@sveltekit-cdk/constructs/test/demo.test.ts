import { App, Stack, StackProps } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { Construct } from 'constructs';
import { SvelteDistribution } from '../src';

class TestStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
    new SvelteDistribution(this, 'site', {
      artifactPath: 'test/artifact',
    });
  }
}

test('demo site', () => {
  const app = new App();
  const stack = new TestStack(app, 'TestStack', {
    env: { region: 'us-east-1' },
  });
  const template = Template.fromStack(stack);
  expect(template).toMatchSnapshot();
});