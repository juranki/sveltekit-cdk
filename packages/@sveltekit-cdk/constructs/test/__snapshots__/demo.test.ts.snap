// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`demo site 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536": Object {
      "DependsOn": Array [
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-us-east-1",
          },
          "S3Key": "2bc265c5e0569aeb24a6349c15bd54e76e845892376515e036627ab0cc70bb64.zip",
        },
        "Handler": "index.handler",
        "Layers": Array [
          Object {
            "Ref": "sitesvelteStaticDeploymentAwsCliLayer7CC15EE2",
          },
        ],
        "Role": Object {
          "Fn::GetAtt": Array [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
            "Arn",
          ],
        },
        "Runtime": "python3.9",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-us-east-1",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-us-east-1",
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "sitesvelteStaticBucketD727A877",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "sitesvelteStaticBucketD727A877",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "cloudfront:GetInvalidation",
                "cloudfront:CreateInvalidation",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "Roles": Array [
          Object {
            "Ref": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A": Object {
      "DependsOn": Array [
        "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
        "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-us-east-1",
          },
          "S3Key": "eb5b005c858404ea0c8f68098ed5dcdf5340e02461f149751d10f59c210d5ef8.zip",
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:PutRetentionPolicy",
                "logs:DeleteRetentionPolicy",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
        "Roles": Array [
          Object {
            "Ref": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "sitedistro85797F9D": Object {
      "Properties": Object {
        "DistributionConfig": Object {
          "CacheBehaviors": Array [
            Object {
              "CachePolicyId": "658327ea-f89d-4fab-a63d-7e88639e58f6",
              "Compress": true,
              "PathPattern": "_app/*",
              "TargetOriginId": "TestStacksitedistroOrigin17EAD3D96",
              "ViewerProtocolPolicy": "redirect-to-https",
            },
            Object {
              "CachePolicyId": "658327ea-f89d-4fab-a63d-7e88639e58f6",
              "Compress": true,
              "PathPattern": "favicon.png",
              "TargetOriginId": "TestStacksitedistroOrigin17EAD3D96",
              "ViewerProtocolPolicy": "redirect-to-https",
            },
            Object {
              "CachePolicyId": "658327ea-f89d-4fab-a63d-7e88639e58f6",
              "Compress": true,
              "PathPattern": "robots.txt",
              "TargetOriginId": "TestStacksitedistroOrigin17EAD3D96",
              "ViewerProtocolPolicy": "redirect-to-https",
            },
            Object {
              "CachePolicyId": "658327ea-f89d-4fab-a63d-7e88639e58f6",
              "Compress": true,
              "PathPattern": "vite-manifest.json",
              "TargetOriginId": "TestStacksitedistroOrigin17EAD3D96",
              "ViewerProtocolPolicy": "redirect-to-https",
            },
          ],
          "DefaultCacheBehavior": Object {
            "AllowedMethods": Array [
              "GET",
              "HEAD",
              "OPTIONS",
              "PUT",
              "PATCH",
              "POST",
              "DELETE",
            ],
            "CachePolicyId": Object {
              "Ref": "sitesvelteDynamicCachePolicy563FD09B",
            },
            "Compress": true,
            "LambdaFunctionAssociations": Array [
              Object {
                "EventType": "origin-request",
                "IncludeBody": true,
                "LambdaFunctionARN": Object {
                  "Ref": "sitesvelteHandlerFnCurrentVersion962DB9CBe8f810a0cf154d38ff87c5f6bdd271f2",
                },
              },
            ],
            "OriginRequestPolicyId": "216adef6-5c7f-47e4-b989-5492eafa07d3",
            "TargetOriginId": "TestStacksitedistroOrigin17EAD3D96",
            "ViewerProtocolPolicy": "redirect-to-https",
          },
          "Enabled": true,
          "HttpVersion": "http2",
          "IPV6Enabled": true,
          "Origins": Array [
            Object {
              "DomainName": Object {
                "Fn::GetAtt": Array [
                  "sitesvelteStaticBucketD727A877",
                  "RegionalDomainName",
                ],
              },
              "Id": "TestStacksitedistroOrigin17EAD3D96",
              "OriginPath": "/static",
              "S3OriginConfig": Object {
                "OriginAccessIdentity": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "origin-access-identity/cloudfront/",
                      Object {
                        "Ref": "sitedistroOrigin1S3Origin569F2CBA",
                      },
                    ],
                  ],
                },
              },
            },
          ],
          "PriceClass": "PriceClass_100",
        },
      },
      "Type": "AWS::CloudFront::Distribution",
    },
    "sitedistroOrigin1S3Origin569F2CBA": Object {
      "Properties": Object {
        "CloudFrontOriginAccessIdentityConfig": Object {
          "Comment": "Identity for TestStacksitedistroOrigin17EAD3D96",
        },
      },
      "Type": "AWS::CloudFront::CloudFrontOriginAccessIdentity",
    },
    "sitesvelteDynamicCachePolicy563FD09B": Object {
      "Properties": Object {
        "CachePolicyConfig": Object {
          "DefaultTTL": 86400,
          "MaxTTL": 31536000,
          "MinTTL": 0,
          "Name": "TestStacksitesvelteDynamicCachePolicyE3E48B97-us-east-1",
          "ParametersInCacheKeyAndForwardedToOrigin": Object {
            "CookiesConfig": Object {
              "CookieBehavior": "all",
            },
            "EnableAcceptEncodingBrotli": false,
            "EnableAcceptEncodingGzip": false,
            "HeadersConfig": Object {
              "HeaderBehavior": "none",
            },
            "QueryStringsConfig": Object {
              "QueryStringBehavior": "none",
            },
          },
        },
      },
      "Type": "AWS::CloudFront::CachePolicy",
    },
    "sitesvelteHandlerFnB2BC5759": Object {
      "DependsOn": Array [
        "sitesvelteHandlerFnServiceRole8CF8DFE0",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-us-east-1",
          },
          "S3Key": "3a41abe58e7b09f29528b6c73eb5b482614f66fcc02329c025810eed3a4a5b7b.zip",
        },
        "Handler": "at-edge-handler.handler",
        "MemorySize": 512,
        "Role": Object {
          "Fn::GetAtt": Array [
            "sitesvelteHandlerFnServiceRole8CF8DFE0",
            "Arn",
          ],
        },
        "Runtime": "nodejs16.x",
        "Timeout": 5,
      },
      "Type": "AWS::Lambda::Function",
    },
    "sitesvelteHandlerFnCurrentVersion962DB9CBe8f810a0cf154d38ff87c5f6bdd271f2": Object {
      "Properties": Object {
        "FunctionName": Object {
          "Ref": "sitesvelteHandlerFnB2BC5759",
        },
      },
      "Type": "AWS::Lambda::Version",
    },
    "sitesvelteHandlerFnLogRetention1ED3C283": Object {
      "Properties": Object {
        "LogGroupName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "/aws/lambda/",
              Object {
                "Ref": "sitesvelteHandlerFnB2BC5759",
              },
            ],
          ],
        },
        "RetentionInDays": 7,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
            "Arn",
          ],
        },
      },
      "Type": "Custom::LogRetention",
    },
    "sitesvelteHandlerFnServiceRole8CF8DFE0": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "edgelambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "sitesvelteStaticBucketD727A877": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "aws-cdk:cr-owned:static:8930388f",
            "Value": "true",
          },
        ],
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "sitesvelteStaticBucketPolicyDE1B2991": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "sitesvelteStaticBucketD727A877",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Principal": Object {
                "CanonicalUser": Object {
                  "Fn::GetAtt": Array [
                    "sitedistroOrigin1S3Origin569F2CBA",
                    "S3CanonicalUserId",
                  ],
                },
              },
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    Object {
                      "Fn::GetAtt": Array [
                        "sitesvelteStaticBucketD727A877",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "sitesvelteStaticDeploymentAwsCliLayer7CC15EE2": Object {
      "Properties": Object {
        "Content": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-us-east-1",
          },
          "S3Key": "5d8d1d0aacea23824c62f362e1e3c14b7dd14a31c71b53bfae4d14a6373c5510.zip",
        },
        "Description": "/opt/awscli/aws",
      },
      "Type": "AWS::Lambda::LayerVersion",
    },
    "sitesvelteStaticDeploymentCustomResourceF5200C3D": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "DestinationBucketKeyPrefix": "static",
        "DestinationBucketName": Object {
          "Ref": "sitesvelteStaticBucketD727A877",
        },
        "DistributionId": Object {
          "Ref": "sitedistro85797F9D",
        },
        "DistributionPaths": Array [
          "/_app/*",
          "/favicon.png",
          "/robots.txt",
          "/vite-manifest.json",
        ],
        "Prune": true,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536",
            "Arn",
          ],
        },
        "SourceBucketNames": Array [
          Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-us-east-1",
          },
        ],
        "SourceObjectKeys": Array [
          "6cb8afe48c633ec732c7f355d9d50dce414248e8864dc9a812026631e8746512.zip",
        ],
        "SystemMetadata": Object {
          "cache-control": "max-age=31536000",
        },
      },
      "Type": "Custom::CDKBucketDeployment",
      "UpdateReplacePolicy": "Delete",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
